src/openqaoa-core/optimizers/training_vqa.py

```diff
@@ -0,0 +1,906 @@
+import os
```

Comment:
perhaps we could call this file 'optimization' instead of 'training'?

Reply:
Since the purpose of the object is to optimize VQAs maybe `optimizing_vqa.py` works too.

I think `optimization_vqa.py` and `optimizing_vqa.py` works better than `training_vqa.py`.

@Q-lds @vishal-ph Quick comments here would help. Thanks.

Reply:
Let's change the name. Training is too ML related, and perhaps a bit strange.

Is this the place where the classical optimisation loop happens? Maybe we can encode that in the title?

`optimisation_loop.py` or something like that?

Reply:
yes this is where the objects that hold the circuits and the classical optimizers come together.

Reply:
`optimization_loop.py` sounds a bit off to me. Since technically, these objects are Optimizers that wrap/contain a VQA Circuit, which then perform optimizations on said circuit.
`optimizing_vqa.py` would be similar to calling it, optimizaing a/the vqa. (A naming scheme I prefer.)

Reply:
Raised issue in #182 continue discussion there.